File: indicator/src/main/java/com/zhpan/indicator/annotation/AIndicatorSlideMode.java
package com.zhpan.indicator.annotation;

import androidx.annotation.IntDef;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import static com.zhpan.indicator.enums.IndicatorSlideMode.NORMAL;
import static com.zhpan.indicator.enums.IndicatorSlideMode.SMOOTH;
import static com.zhpan.indicator.enums.IndicatorSlideMode.WORM;


/**
 * <pre>
 *   Created by zhangpan on 2019-10-18.
 *   Description:
 * </pre>
 */
@IntDef({NORMAL, SMOOTH,WORM})
@Retention(RetentionPolicy.SOURCE)
@Target({ElementType.PARAMETER,ElementType.FIELD})
public @interface AIndicatorSlideMode {
}

File: indicator/src/main/java/com/zhpan/indicator/annotation/AIndicatorSlideMode.kt
package com.zhpan.indicator.annotation

import androidx.annotation.IntDef
import com.zhpan.indicator.enums.IndicatorSlideMode.Companion.COLOR
import com.zhpan.indicator.enums.IndicatorSlideMode.Companion.NORMAL
import com.zhpan.indicator.enums.IndicatorSlideMode.Companion.SCALE
import com.zhpan.indicator.enums.IndicatorSlideMode.Companion.SMOOTH
import com.zhpan.indicator.enums.IndicatorSlideMode.Companion.WORM

/**
 * <pre>
 * Created by zhangpan on 2019-10-18.
 * Description:
</pre> *
 */
@IntDef(NORMAL, SMOOTH, WORM, COLOR, SCALE)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FIELD)
annotation class AIndicatorSlideMode
